cli/cli.go

```diff
@@ -0,0 +1,168 @@
+// Package cli provides boilerplate code required to generate plugin executable.
+// At high level it performs following tasks
+// 1. Validate command arguments
+// 2. Read and unmarshal input
+// 3. Execute relevant plugin functions
+// 4. marshals output
+
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/notaryproject/notation-plugin-framework-go/internal/slices"
+	"github.com/notaryproject/notation-plugin-framework-go/log"
+	"github.com/notaryproject/notation-plugin-framework-go/plugin"
+)
+
+type Cli struct {
+	syntax string
+	pl     plugin.Plugin
+}
+
+// New creates a new Cli using given plugin
+func New(pl plugin.Plugin) *Cli {
+	return &Cli{pl: pl}
+}
+
+// Execute is main controller that reads/validates commands, parses input, executes relevant plugin functions
+// and returns corresponding output.
+func (c Cli) Execute(ctx context.Context, args []string) {
```

Comment:
Any reason for using value receiver for `CLI` instead of pointer receiver?

Reply:
Nop, updated to use pointer receiver.